app/assets/javascripts/mpa/api/pept2data/Response.ts

```diff
@@ -0,0 +1,47 @@
+export type PeptideData = 
+{
+    lca: number, 
+    lineage: [number], 
+    fa: 
+    {
+        counts: 
+        {
+            all: number,
+            EC: number;
+            GO: number;
+        },
+        data: any
+    }
+}
+
+type Response = Map<string, PeptideData>;
+
+export class Pept2DataResponse
+{
+    public response: Response;
+
+    constructor(response: Response = new Map<string, PeptideData>())
+    {
+        this.response = response;
+    }
+
+    setPeptideData(peptide: string, peptideData: PeptideData)
+    {
+        this.response.set(peptide, peptideData);
+    }
+
+    GetPeptideData(peptide: string) : Readonly<PeptideData>
```

Comment:
Minor remark: use consistent casing for getters and setters (preferably camelCase with a lowercase first letter)